library/kotlin/src/io/envoyproxy/envoymobile/RetryPolicy.kt

```diff
@@ -0,0 +1,26 @@
+package io.envoyproxy.envoymobile
+
+/**
+ * Specifies how a request may be retried, containing one or more rules.
+ *
+ * @param maxRetryCount Maximum number of retries that a request may be performed.
+ * @param retryOn Whitelist of rules used for retrying.
+ * @param perRetryTimeoutMs Timeout (in milliseconds) to apply to each retry.
+ */
+data class RetryPolicy(
+    val maxRetryCount: Int,
+    val retryOn: List<RetryRule>,
+    val perRetryTimeoutMs: Long?
+)
+
+/**
+ * These are retry rules specified in Envoy's router filter.
+ * @see <a href="https://www.envoyproxy.io/docs/envoy/latest/configuration/http_filters/router_filter#x-envoy-retry-on">x-envoy-retry-on</a>
+ */
+enum class RetryRule {
+  ALL_5XX,
```

Comment:
These don't quite match what's defined in Swift. Do you want to update Kotlin or should we change Swift's? https://github.com/lyft/envoy-mobile/blob/master/library/swift/src/RetryPolicy.swift#L7

Personal preference is to update Kotlin since the Swift statuses are reflective of Envoy upstream

Reply:
I agree with mapping to Envoy's options directly.

Reply:
I made a mental note to update this when I saw what you did in your PR @rebello95 😬